### PR TITLE
solc: add tracing

### PR DESCRIPTION
**Pull request by gakonst**
adds tracing to solc so we know what takes how long etc., good for figuring out places for optimization with `RUST_LOG=ethers_solc=trace`

### CODE REVIEW

File: ethers-solc/src/lib.rs [hunk 1]
```diff
@@ -273,7 +311,9 @@ impl<Artifacts: ArtifactOutput> Project<Artifacts> {
         let input = CompilerInput::with_sources(sources)
             .normalize_evm_version(&solc.version()?)
             .with_remappings(self.paths.remappings.clone());
+        tracing::trace!("calling solc");
         let output = solc.compile(&input)?;
+        tracing::trace!("done calling solc");
```
Comment (mattsse) [inline ethers-solc/src/lib.rs:316]:
```suggestion
        tracing::trace!("compiled input, output has error: {}", output.has_error());
```

File: ethers-solc/src/lib.rs [hunk 2]
```diff
@@ -228,37 +255,48 @@ impl<Artifacts: ArtifactOutput> Project<Artifacts> {
...
         let mut path_to_source_name = HashMap::new();
 
+        tracing::trace!("resolving libraries");
         for (import, (source, path)) in self.resolved_libraries(&sources)? {
             // inserting with absolute path here and keep track of the source name <-> path mappings
             sources.insert(path.clone(), source);
             path_to_source_name.insert(path.clone(), import.clone());
             source_name_to_path.insert(import, path);
         }
+        tracing::trace!("done resolving libraries");
```
Comment (mattsse) [inline ethers-solc/src/lib.rs:274]:
```suggestion
        tracing::trace!("resolved libraries");
```

File: ethers-solc/src/lib.rs [hunk 3]
```diff
@@ -204,21 +221,31 @@ impl<Artifacts: ArtifactOutput> Project<Artifacts> {
             let entry = sources_by_version.entry(solc).or_insert_with(BTreeMap::new);
             entry.insert(path.clone(), source);
         }
+        // tracing::trace!("done");
 
         let mut compiled =
             ProjectCompileOutput::with_ignored_errors(self.ignored_error_codes.clone());
 
         // run the compilation step for each version
+        tracing::trace!("compiling all versions");
```
Comment (mattsse) [inline ethers-solc/src/lib.rs:230]:
```suggestion
        tracing::trace!("compiling sources with viable solc versions");
```

File: ethers-solc/src/lib.rs [hunk 4]
```diff
@@ -204,21 +221,31 @@ impl<Artifacts: ArtifactOutput> Project<Artifacts> {
...
+                tracing::trace!("missing version, redownloading...");
                 Solc::blocking_install(version)?;
+                tracing::trace!("done.");
             }
             // once matched, proceed to compile with it
+            tracing::trace!("compiling_with_version");
             compiled.extend(self.compile_with_version(&solc, sources)?);
+            tracing::trace!("done compiling_with_version");
         }
+        tracing::trace!("done compiling all versions");
```
Comment (mattsse) [inline ethers-solc/src/lib.rs:248]:
```suggestion
        tracing::trace!("compiled sources with viable solc versions");
```

**Review by mattsse — COMMENTED** [review]
some suggestions

File: ethers-solc/src/lib.rs [hunk 5]
```diff
@@ -107,13 +110,18 @@ impl<Artifacts: ArtifactOutput> Project<Artifacts> {
             }
         }
 
+        tracing::trace!("writing files to disk");
```
Comment (gakonst) [inline ethers-solc/src/lib.rs:113]:
```suggestion
```

File: ethers-solc/src/lib.rs [hunk 6]
```diff
@@ -107,13 +110,18 @@ impl<Artifacts: ArtifactOutput> Project<Artifacts> {
...
         if let Some(cache_dir) = self.paths.cache.parent() {
             fs::create_dir_all(cache_dir)?
         }
-        cache.write(&self.paths.cache)
+        cache.write(&self.paths.cache)?;
+        tracing::trace!("files written");
```
Comment (gakonst) [inline ethers-solc/src/lib.rs:118]:
```suggestion
```

**Pull request merged**